rootston/cursor.c

```diff
@@ -228,6 +250,20 @@ static void roots_cursor_press_button(struct roots_cursor *cursor,
 	struct wlr_surface *surface = desktop_surface_at(desktop,
 			lx, ly, &sx, &sy, &view);
 
+	if (!is_touch && cursor->pointer_view && cursor->pointer_view != view) {
```

Comment:
This is racy: there's no guarantee there will be a motion event on this view before a button event.

What about saving on button press the view on which the press occurred?